game/src/main/java/net/demilich/metastone/game/logic/GameLogic.java

```diff
@@ -1927,7 +1953,7 @@ private void handleFrozen(Actor actor) {
 		if (!actor.hasAttribute(Attribute.FROZEN)) {
 			return;
 		}
-		if (actor.getAttributeValue(Attribute.NUMBER_OF_ATTACKS) >= actor.getMaxNumberOfAttacks()) {
+		if (actor.getAttributeValue(Attribute.NUMBER_OF_ATTACKS) >= actor.getMaxNumberOfAttacks() && !actor.hasAttribute(Attribute.DONT_UNFREEZE)) {
```

Comment:
It's fine, but maybe it should really be a trigger that refreezes it. Otherwise, this is going to have a lot of subtle effect copying bugs.